src/transformers/file_utils.py

```diff
@@ -806,3 +806,22 @@ def wrapper(*args, **kwargs):
             raise ImportError(f"Method `{func.__name__}` requires TF.")
 
     return wrapper
+
+
+class ModelOutput:
+    """
+    Base class for all model outputs as dataclass. Has a ``__getitem__`` (to make it behave like a ``namedtuple``) that
+    will ignore ``None`` in the attributes.
+    """
+
+    def to_tuple(self):
+        return tuple(getattr(self, f) for f in self.__dataclass_fields__.keys() if getattr(self, f, None) is not None)
+
+    def to_dict(self):
+        return {f: getattr(self, f) for f in self.__dataclass_fields__.keys() if getattr(self, f, None) is not None}
+
+    def __getitem__(self, i):
+        return self.to_tuple()[i]
```

Comment:
Shall we go up the way to test if `i` is an integer or a string?

Reply:
That would be fancy!

Reply:
Oh clever! That way this will behave as a namedtuple **and**  a dict. I like it!